File: CSES/Sorting and Searching/Restaurant Customers/restaurantCostumers.cpp
#include <bits/stdc++.h>
using namespace std;

#define FOR(i,a,b) for (int i = a; i < b; i++)
typedef vector<int> vi;
#define sortv(v) sort((v).begin(), (v).end())

int main() {
  int n; cin >> n;
  vi e(n), s(n);

  FOR(i,0,n){
    cin >> e[i] >> s[i];
  }

  sortv(e);
  sortv(s);

  int c = 0, i = 0, j = 0, sol = 0;

  while(i < n && j < n){
    if (e[i] < s[j]){
      c++;
      i++;
    }
    else{
      c--;
      j++;
    }

    sol = max(sol, c);
  }
  
  cout << sol;
}

File: CSES/Sorting and Searching/Restaurant Customers/restaurantCostumers.py
i1 = int(input())
 
e = []
s = []
 
for i in range (i1):
  a, b = input().split()
  e.append(int(a))
  s.append(int(b))
 
e.sort()
s.sort()
 
i = 0
j = 0
m = 0
c = 0
 
while i < i1 and j < i1:
  if e[i] < s[j]:
    c += 1
    i += 1
  else:
    c -= 1
    j += 1
    
  m = max(m, c)
 
print(m)